### ISSUE TITLE
wasm support?

### ISSUE THREAD

**Issue by zhangzqs** [maxwase/button-driver#8]
Is it possible to port it to run on the wasm platform?
wasm in browser don't support `Instant::now()`.

**Comment by maxwase** [comment]
That's an interesting question, I didn't consider WASM as a platform for this crate, however it seems that it is possible, so why not 😄

As you have already spotted, there is an issue for an external timer trait, honestly, I don't have good ideas on how to define one straight away, however I didn't think about it much.

Could you please share an example on how you would like to use this crate on WASM platform and how you would implement `Instant` on it if it misses one? That would help to consider needed aspects for the trait.

**Comment by zhangzqs** [comment]
> That's an interesting question, I didn't consider WASM as a platform for this crate, however it seems that it is possible, so why not 😄
> 
> As you have already spotted, there is an issue for an external timer trait, honestly, I don't have good ideas on how to define one straight away, however I didn't think about it much.
> 
> Could you please share an example on how you would like to use this crate on WASM platform and how you would implement `Instant` on it if it misses one? That would help to consider needed aspects for the trait.

Actually, we don't rely on absolute time, we only need to rely on the relative time since the Button was new, so we can actually just use core::time::Duration instead of Instant.

**Comment by zhangzqs** [comment]
I tried to file a draft PR to change this, but noticed that the State enum and raw_state function are public, which means some incompatible changes have occurred to the external API.

https://github.com/maxwase/button-driver/pull/9/files

**Comment by maxwase** [comment]
It would be great if you could try to implement your use case using #10

**Comment by zhangzqs** [comment]
Yes, now it works for my use case, and I added an example running on the browser wasm.

https://github.com/maxwase/button-driver/pull/16
